data/block/blockV2.go

```diff
@@ -0,0 +1,617 @@
+	if hv2 == nil {
+		return data.ErrNilPointerReceiver
+	}
+	if hv2.Header == nil {
+		return data.ErrNilPointerDereference
+	}
+
+	hv2.Header.ChainID = chainID
+	return nil
+}
+
+// SetSoftwareVersion sets the software version of the header
+func (hv2 *HeaderV2) SetSoftwareVersion(version []byte) error {
+	if hv2 == nil {
+		return data.ErrNilPointerReceiver
+	}
+	if hv2.Header == nil {
+		return data.ErrNilPointerDereference
+	}
+
+	hv2.Header.SoftwareVersion = version
+	return nil
+}
+
+// SetTimeStamp sets header timestamp
+func (hv2 *HeaderV2) SetTimeStamp(ts uint64) error {
+	if hv2 == nil {
+		return data.ErrNilPointerReceiver
+	}
+	if hv2.Header == nil {
+		return data.ErrNilPointerDereference
+	}
+
+	hv2.Header.TimeStamp = ts
+	return nil
+}
+
+// SetAccumulatedFees sets the accumulated fees in the header
+func (hv2 *HeaderV2) SetAccumulatedFees(value *big.Int) error {
+	if hv2 == nil {
+		return data.ErrNilPointerReceiver
+	}
+	if hv2.Header == nil {
+		return data.ErrNilPointerDereference
+	}
+	if hv2.Header.AccumulatedFees == nil {
+		hv2.Header.AccumulatedFees = big.NewInt(0)
+	}
+
+	hv2.Header.AccumulatedFees.Set(value)
+	return nil
+}
+
+// SetDeveloperFees sets the developer fees in the header
+func (hv2 *HeaderV2) SetDeveloperFees(value *big.Int) error {
+	if hv2 == nil {
+		return data.ErrNilPointerReceiver
+	}
+	if hv2.Header == nil {
+		return data.ErrNilPointerDereference
+	}
+	if hv2.Header.DeveloperFees == nil {
+		hv2.Header.DeveloperFees = big.NewInt(0)
+	}
+	hv2.Header.DeveloperFees.Set(value)
+	return nil
+}
+
+// SetTxCount sets the transaction count of the block associated with this header
+func (hv2 *HeaderV2) SetTxCount(txCount uint32) error {
+	if hv2 == nil {
+		return data.ErrNilPointerReceiver
+	}
+	if hv2.Header == nil {
+		return data.ErrNilPointerDereference
+	}
+
+	hv2.Header.TxCount = txCount
+	return nil
+}
+
+// SetShardID sets header shard ID
+func (hv2 *HeaderV2) SetShardID(shId uint32) error {
+	if hv2 == nil {
+		return data.ErrNilPointerReceiver
+	}
+	if hv2.Header == nil {
+		return data.ErrNilPointerDereference
+	}
+
+	hv2.Header.ShardID = shId
+	return nil
+}
+
+// GetMiniBlockHeadersWithDst as a map of hashes and sender IDs
+func (hv2 *HeaderV2) GetMiniBlockHeadersWithDst(destId uint32) map[string]uint32 {
+	if hv2 == nil || hv2.Header == nil {
+		return nil
+	}
+
+	hashDst := make(map[string]uint32)
+	for _, val := range hv2.Header.MiniBlockHeaders {
+		if val.ReceiverShardID == destId && val.SenderShardID != destId {
+			hashDst[string(val.Hash)] = val.SenderShardID
+		}
+	}
+	return hashDst
+}
+
+// GetOrderedCrossMiniblocksWithDst gets all cross miniblocks with the given destination shard ID, ordered in a
+// chronological way, taking into consideration the round in which they were created/executed in the sender shard
+func (hv2 *HeaderV2) GetOrderedCrossMiniblocksWithDst(destId uint32) []*data.MiniBlockInfo {
+	if hv2 == nil || hv2.Header == nil {
```

Comment:
Here also (and all getters/setters containing for loops) could have reused the code from hv2.Header